compute_sdk/globus_compute_sdk/sdk/executor.py

```diff
@@ -348,7 +349,14 @@ def register_function(
 
         log.debug("Function not registered. Registering: %s", fn)
         func_register_kwargs.pop("function", None)  # just to be sure
-        reg_kwargs = {"function_name": fn.__name__, "container_uuid": self.container_id}
+        reg_kwargs = {
+            "function_name": fn.__name__,
+            "container_uuid": self.container_id,
+            "metadata": {
+                "python_version": sys.version.split()[0],
```

Comment:
The first chunk before the string seems to be always the major.minor.micro, though technically sys.version_info[major.minor.micro] is more 'precise'.  However this is much simpler and unlikely to change so probably a good choice.

Reply:
On second thought, I'll use `platform.python_version()` instead since it's explicitly recommended in the [Python docs](https://docs.python.org/3/library/sys.html#sys.version).